[DO NOT MERGE] Add a test to prove how to use protobuf nullability

gortiz opened the pull request:
This PR adds a test to verify the correct nullability read on `ProtoBufRecordExtractor`.

The PR is not yet ready to merge because for some reason the descriptor is using v2 instead of 3.

cc @swaminathanmanish

swaminathanmanish commented on pinot-plugins/pinot-input-format/pinot-protobuf/src/main/java/org/apache/pinot/plugin/inputformat/protobuf/ProtoBufRecordExtractor.java:
```diff
@@ -62,7 +62,7 @@ private Object getFieldValue(Descriptors.FieldDescriptor fieldDescriptor, Messag
     // if its explicitly set to a proto default or not been set at all i.e hasField() returns false
     // and we would use null.
-    if (fieldDescriptor.isRepeated() || !fieldDescriptor.isOptional() || message.hasField(fieldDescriptor)) {
+    if (fieldDescriptor.isRepeated() || !fieldDescriptor.hasOptionalKeyword() || message.hasField(fieldDescriptor)) {
```
This is exactly what my PR changed - from hasOptionalKeyword to isOptional

https://github.com/apache/pinot/pull/11682

API is deprecated
https://github.com/protocolbuffers/protobuf/commit/d6157f7c7ec78a6942cfd6d96d2e0b40e007ae0d

gortiz replied:
It is deprecated in latest versions of protobuf, but:
- It is not deprecated in the version we use
- There is no alternative to this method in the latest protobuf version.

Theoretically we should is `isPresent`, but it is not honoring its javadoc. You can verify that by changing this to `isPresent` and see that it returns false for the test added here.

gortiz replied:
It seems isPresent is not bug, but I was unable to correctly configure protobuf model programmatically.

gortiz commented:
I've changed the code to:
- Recover the logic in master (use isOptional)
- Add more tests
  - Renamed `ProtoBufRecordExtractorSimpleTest` as `ProtoBufRecordExtractorProgrammaticTest`
  - Add `ProtoBufRecordExtractorLowLevelTest` which runs tests using the same protocol definition used by `ProtoBufRecordExtractorTest` but doesn't include all the error prone magic in `AbstractRecordExtractorTest`.

I think `ProtoBufRecordExtractorLowLevelTest` is easier to read. There are 4 test cases:
1. When the field is set to some non default value
2. When the field is set to the default value
3. When the field is not set
4. When the field is clear (which should be semantically equal to 3, but I wanted to be sure.

It is easier and more useful to read the DataProviders than the tests.

As you can see, both `ProtoBufRecordExtractorProgrammaticTest` and `ProtoBufRecordExtractorLowLevelTest` have some test that fail. All these fails can be fixed by recovering Shen's code where we used `isOptionalKeyword` instead of `isOptional`
